Month 11/step 1

gudrum1983 opened the pull request:
Доброго времени суток ревьювер!
Геннадий или Данил, если это кто-нибудь из вас, вам отдельный пламенный привет!

 P.S. По итогу я так и не смогла придумать как лучше хранить историю :) То есть у нас есть элементы которые меняют своё состояние (это я реализовывала в начале), а можно хранить историю всего "объекта" указывая указателями какой объект меняется и как (до этого я дошла только на последних алгоритмах). Если есть пример такой работы или иная информация, буду признательна за ссылки :)

Приятного просмотра :)

gudrum1983 committed:
feat: Создание новой ветки для спринта 11 - Алгоритмы и структура данных // Creating a new branch for Sprint 11 - Algorithms and Data Structures

gudrum1983 committed:
feat: создать разметку страницы  string.tsx// Create markup for the page string.tsx

gudrum1983 committed:
feat: создать заполнение стейта по таймингу в   string.tsx// Create state filling by timing in string.tsx

gudrum1983 committed:
feat: подключить библиотеку uuid// сonnect the uuid library

gudrum1983 committed:
feat:
добавить заготовки алгоритмов очереди, стека//
add queue and stack algorithm templates

gudrum1983 committed:
feat:
написать функцию реверса и компонент строка, добавить компонент для пошагового отображения//
Write a reverse function and String component, add a step-by-step display component

gudrum1983 committed:
feat:
написать функцию реверса и компонент строка, добавить компонент для пошагового отображения//
Write a reverse function and String component, add a step-by-step display component

gudrum1983 committed:
feat:
написать функцию фибоначчи//
Write a Fibonacci function

gudrum1983 committed:
refactor:
начать рефакторинг алгоритма переворота//
Start refactoring the reversal algorithm

gudrum1983 committed:
refactor:
начать рефакторинг алгоритма переворота//
Start refactoring the reversal algorithm

gudrum1983 committed:
refactor:
продолжить рефакторинг алгоритма переворота//
Continue refactoring the reversal algorithm

gudrum1983 committed:
refactor:
продолжить рефакторинг алгоритма переворота часть 2//
Continue refactoring the reversal algorithm part 2

gudrum1983 committed:
refactor:
закончить основной рефакторинг алгоритма переворота строки//
finish the main refactoring of the string reversal algorithm

gudrum1983 committed:
feat:
добавить конфетти//
add confetti

gudrum1983 committed:
feat:
создать функционал нахождения фибоначчи//
create functionality for finding Fibonacci numbers

gudrum1983 committed:
refactor:
перенести файл reverse.ts и sorting-page.tsx, подготовить файлы к удвлению пакета uuid//
move the files reverse.ts and sorting-page.tsx, prepare the files for removing the uuid package

gudrum1983 committed:
refactor:
удаление пакета uuid//
remove packages uuid and @types/uuid

gudrum1983 committed:
refactor:
создать начальный лейаут страницы сортировки с обраоткой событий кнопок//
create the initial layout of the sorting page with button event handling

gudrum1983 committed:
feat:
закончить начальный лейаут страницы сортировки с обработкой событий кнопок, добавить функцию создания рандомного массива чисел//
finish the initial layout of the sorting page with button event handling, add a function for creating a random array of numbers

gudrum1983 committed:
refactor:
провести рефакторинг функции создания рандомного массива чисел//
refactor the function for creating a random array of numbers

gudrum1983 committed:
refactor:
провести рефакторинг utils.ts//
refactor the utils.ts

gudrum1983 committed:
refactor:
удалить опциональность 'state' в типе элемента строки//
remove optionality of 'state' in the row element type

gudrum1983 committed:
feat:
закончить функционал сортировки//
finish the sorting functionality

gudrum1983 committed:
refactor:
проверить нейминг констант, отрефакторить сортировку, пересмотреть типы элементов//
check the naming of constants, refactor the sorting, and review the types of elements

gudrum1983 committed:
feat:
создать функционал алгоритма стека, создать useForm и доработать инпут для форвард реф//
create stack algorithm functionality, create useForm and refine input for forward ref

gudrum1983 committed:
feat:
создать страницу алгоритма "Очередь" и функционал добавления в очередь//
сreate a 'Queue' algorithm page and functionality to add to the queue

gudrum1983 committed:
feat:
создать остальной функционал алгоритма работы с очередью, изменить работу снепшотов у очереди, добавить дополнительный компонент пошаговой отрисовки //
create the rest of the queue algorithm functionality, change the queue snapshots handling, add an additional step-by-step rendering component

gudrum1983 committed:
feat:
создать частичный лейаут страницы связного списка, создать классы и типы для работы со списком//
create a partial layout of the linked list page, create classes and types for working with the list

gudrum1983 committed:
feat:
дописать функционал списка//
complete the list functionality

gudrum1983 committed:
feat:
начать рефакторинг списка//
start refactoring the list

gudrum1983 committed:
refactor:
рефакторинг страницы списка, изменить функционал отключения активности филдсетов//
refactoring of the list page, change the functionality of disabling fieldsets

gudrum1983 committed:
refactor:
рефакторинг типов компонента step-by-step-display3.tsx списка и очереди, для совместного использования//
refactoring of types in step-by-step-display3.tsx component for the list and queue to be shared

gudrum1983 committed:
refactor:
рефакторинг fibonacci и string//
refactoring fibonacci and string

gudrum1983 committed:
refactor:
рефакторинг sorting//
refactoring sorting

gudrum1983 committed:
refactor:
рефакторинг stack//
refactoring stack

gudrum1983 committed:
refactor:
рефакторинг css//
refactoring css

gudrum1983 committed:
refactor:
рефакторинг переменных, типов и утилс //
refactoring of variables, types and utils

gudrum1983 committed:
refactor:
рефакторинг переменных, типов и утилс //
refactoring of variables, types and utils

gennady-bars commented on src/useForm.ts:
```diff
@@ -0,0 +1,20 @@
+  setValues: (state: T) => void;
+}
+
+export default function useForm<T>(initialState: T): TUseForm<T> {
```
Отлично, что используете `useForm`

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:


* Отлично, что почти нет типов `any`
* Отлично типизировали проект
* Хорошо сделаны классы
* Отлично, что используете `useForm`
* Хорошо сделали визуализацию
* Прикольные конфетти

но есть некоторые недочеты:

* Файл `app.test.js` нужно удалить, так как он не относится к проекту
* `app.css` нужно сделать модулем `app.module.css`, а в компоненте использовать модульные классы, а не строки 
* В консоли ошибка `key` на странице `Связный список`  https://disk.yandex.ru/i/p8fTRb4-a1aJrg 
* Скрин https://disk.yandex.ru/i/TB5BVthR-y6v3w По заданию все кнопки сабмита должны быть **неактивными**, если  в инпуте ничего нет или в нем невалидные данные.

gudrum1983 committed:
fix:
исправление ошибки ключей в связном списке, восстановление дизейблов в стеке, убрать файл с тестом, изменить css на модуль в APP //
fixing key errors in linked list, restoring disables in stack, remove test file, change css to module in APP

gudrum1983 committed:
feat: gh-page   "homepage": "./"

gudrum1983 committed:
feat: gh-page   "homepage": "./"

gudrum1983 committed:
feat: gh-page   "homepage": "./"

gudrum1983 committed:
feat: gh-page   "homepage": "./"

gudrum1983 committed:
feat: gh-page

gudrum1983 committed:
feat:"homepage": "https://gudrum1983.github.io/algososh/"

gudrum1983 committed:
fix:     "predeploy": "npm run build",
         "deploy": "gh-pages -d dist"

gudrum1983 committed:
fix:     "predeploy": "npm run build",
         "deploy": "gh-pages -d build"

gudrum1983 committed:
fix:  <BrowserRouter basename='/algososh/'>

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
